snapdenv/useragent_test.go

```diff
@@ -64,6 +73,20 @@ func (s *UASuite) TestUserAgent(c *C) {
 	c.Check(strings.Contains(ua, "devmode"), Equals, true)
 }
 
+func (s *UASuite) TestUserAgentWSL(c *C) {
+	restore := mockWsl(false)
+	snapdenv.SetUserAgentFromVersion("10", nil)
+	ua := snapdenv.UserAgent()
+	c.Check(strings.Contains(ua, "wsl"), Equals, false)
+	restore()
```

Comment:
we tend to do this via `defer restore()`

Reply:
That looks cleaner indeed. Fixed.